Description: Fix gpfdist gprintln calls
 The function gprintln function specifies that its arguments should be
 type-checked against a format string. If the second argument to gprintln isn't
 a format string, then GCC will emit a warning/error if -Wformat is used.
Author: Bradford D. Boyle <bradford.d.boyle@gmail.com>
Forwarded: no
Last-Update: 2023-02-04

--- a/src/bin/gpfdist/gpfdist.c
+++ b/src/bin/gpfdist/gpfdist.c
@@ -4617,7 +4617,7 @@
 	if (!r->zstd_cctx)
 	{
 		snprintf(r->zstd_error, r->zstd_err_len, "Creating compression context failed, out of memory.");
-		gprintln(NULL, r->zstd_error);
+		gprintln(NULL, "%s", r->zstd_error);
 		return -1;
 	}
 
@@ -4625,7 +4625,7 @@
 	if (ZSTD_isError(init_result)) 
 	{
 		snprintf(r->zstd_error, r->zstd_err_len, "Creating compression context initialization failed, error is %s.", ZSTD_getErrorName(init_result));
-		gprintln(NULL, r->zstd_error);
+		gprintln(NULL, "%s", r->zstd_error);
 		return -1;
 	}
 
@@ -4640,7 +4640,7 @@
 			if (ZSTD_isError(res))
 			{
 				snprintf(r->zstd_error, r->zstd_err_len, "Compression failed, error is %s.", ZSTD_getErrorName(res));
-				gprintln(NULL, r->zstd_error);
+				gprintln(NULL, "%s", r->zstd_error);
 				return -1;
 			}
 			offset += bout.pos;
@@ -4654,7 +4654,7 @@
 	if (remainingToFlush)
 	{
 		snprintf(r->zstd_error, r->zstd_err_len, "Compression failed, error is not fully flushed.");
-		gprintln(NULL, r->zstd_error);
+		gprintln(NULL, "%s", r->zstd_error);
 		return -1;
 	}
 	offset += output.pos;
